File: dotnet/SpiderEngine/Abstract/IContentParser.cs
namespace SpiderEngine.Abstract;

using SpiderEngine.Model;
using SpiderEngine.Parser.Content;

internal interface IContentParser : IContextProvider
{
    void Parse(string content, string patten, IEnumerable<SpiderKeyPathPair> value);

    public static IContentParser Create(SpiderParserType type) => type switch
    {
        SpiderParserType.Json => new JsonContentParser(),
        SpiderParserType.Regex => new RegexContentParser(),
        _ => throw new NotImplementedException(),
    };
}

File: android/src/main/kotlin/com/itoken/team/wutspider/interfaces/IContentParser.kt
package com.itoken.team.wutspider.interfaces

import com.itoken.team.wutspider.impl.parser.content.JsonContentParser
import com.itoken.team.wutspider.impl.parser.content.RegexContentParser
import com.itoken.team.wutspider.model.SpiderKeyPathPair
import com.itoken.team.wutspider.model.SpiderParserType

interface IContentParser : IContextProvider {

    fun parse(content: String, pattern: String?, value: List<SpiderKeyPathPair>)

    companion object {

        fun create(type: SpiderParserType): IContentParser =
            when (type) {
                SpiderParserType.JSON -> JsonContentParser()
                SpiderParserType.REGEX -> RegexContentParser()
                else -> throw NotImplementedError("unsupported content parser type: $type")
            }

    }

}